backend/xivdpscalc/pps/sample_sch_rotation.py

```diff
@@ -0,0 +1,71 @@
+""" Sample implementations of scholar rotations """
+
+from xivdpscalc.pps.rotation import SchRotation
+from xivdpscalc.pps.sch_action import SchAction, SchEffect, SchResource
+from xivdpscalc.types import ElapsedTime
+
+class FixedSchRotation(SchRotation): #pylint: disable=too-few-public-methods
+    """
+    A Rotation that returns actions in a set order and uses a default action once it iterates through
+    """
+
+    def __init__(self, rotation: list[SchAction], default_action: SchAction):
+        self.rotation = rotation
+        self.index = 0
+        self.default_action = default_action
+
+    def get_action(self, current_time: ElapsedTime, cooldowns: dict[SchAction, float],
+                   remaining_effect_durations: dict[SchEffect, float], resources: dict[SchResource, int]) -> SchAction:
+        """
+        Returns a SchAction based on the fixed rotation
+        """
+        if self.index < len(self.rotation):
+            self.index += 1
+            return self.rotation[self.index - 1]
+        return self.default_action
+
+class SampleSchRotation(SchRotation): #pylint: disable=too-few-public-methods
+    """
+    An example rotation exibiting more dynamic decision rotation generation.
+    This is not meant to be a perfect replication of a good sch rotation, this is primarily for testing and demonstration.
+    """
+
+    def __init__(self):
+        self.rotation = [SchAction.BIOLYSIS, SchAction.AETHERFLOW, SchAction.ENERGYDRAIN,
+                         SchAction.BROIL3, SchAction.BROIL3, SchAction.RUIN2, SchAction.CHAINSTRATAGEM, SchAction.ENERGYDRAIN,
+                         SchAction.BROIL3, SchAction.RUIN2, SchAction.ENERGYDRAIN, SchAction.SWIFTCAST, SchAction.BROIL3,
+                         SchAction.DISSIPATION, SchAction.ENERGYDRAIN, SchAction.BROIL3, SchAction.RUIN2, SchAction.ENERGYDRAIN]
+        self.index = 0
+
+    def get_action(self, current_time: ElapsedTime, cooldowns: dict[SchAction, float],
```

Comment:
not reviewing this function until the if/elif spaghetti is resolved, my brain hurty. remind me to review the rotation logic after you've fixed it.